src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceService.kt

```diff
@@ -32,22 +33,50 @@ class LicenceService(
       curfewAddress = getAddress(curfew, cas2Referral, proposedAddress),
       firstNightCurfewHours = curfew?.firstNight?.let { transformToModelFirstNight(it) },
       // curfewHours referred to as curfewTimes in CVL as going forward a more suitable name and had to distinguish between the two different curfew data formats
-      curfewTimes = curfew?.let { transformToModelCurfewTimes(it.curfewHours) },
+      curfewTimes = if (checkForNullValues(curfew?.curfewHours)) curfew?.let { transformToModelCurfewTimes(it.curfewHours) } else null,
     )
   }
 
+  fun checkForNullValues(curfewHours: CurfewHours?): Boolean {
+    val missingTimes = mutableListOf<String>()
+    if (curfewHours?.mondayFrom == null) { missingTimes += "mondayFrom" }
+    if (curfewHours?.mondayUntil == null) { missingTimes += "mondayUntil" }
+    if (curfewHours?.tuesdayFrom == null) { missingTimes += "tuesdayFrom" }
+    if (curfewHours?.tuesdayUntil == null) { missingTimes += "tuesdayUntil" }
+    if (curfewHours?.wednesdayFrom == null) { missingTimes += "wednesdayFrom" }
+    if (curfewHours?.wednesdayUntil == null) { missingTimes += "wednesdayUntil" }
+    if (curfewHours?.thursdayFrom == null) { missingTimes += "thursdayFrom" }
+    if (curfewHours?.thursdayUntil == null) { missingTimes += "thursdayUntil" }
+    if (curfewHours?.fridayFrom == null) { missingTimes += "fridayFrom" }
+    if (curfewHours?.fridayUntil == null) { missingTimes += "fridayUntil" }
+    if (curfewHours?.saturdayFrom == null) { missingTimes += "saturdayFrom" }
+    if (curfewHours?.saturdayUntil == null) { missingTimes += "saturdayUntil" }
+    if (curfewHours?.sundayFrom == null) { missingTimes += "sundayFrom" }
+    if (curfewHours?.sundayUntil == null) { missingTimes += "sundayUntil" }
+
+    if (missingTimes.isNotEmpty()) {
+      log.info("Missing curfew time(s) for $missingTimes")
+      return false
+    }
+    return true
+  }
+
   fun getAddress(curfew: Curfew?, cas2Referral: Cas2Referral?, proposedAddress: ProposedAddress?): ModelCurfewAddress? {
     val isCurfewApprovedPremisesRequired = curfew?.approvedPremises?.required == Decision.YES
     val isCas2ApprovedPremisesRequired = cas2Referral?.bassAreaCheck?.approvedPremisesRequiredYesNo == Decision.YES
     val isCas2Requested = cas2Referral?.bassRequest?.bassRequested == Decision.YES
     val isCas2Accepted = cas2Referral?.bassOffer?.bassAccepted == OfferAccepted.YES
 
     val address = when {
-      isCurfewApprovedPremisesRequired && !isCas2ApprovedPremisesRequired -> curfew.approvedPremisesAddress
-      isCas2ApprovedPremisesRequired -> cas2Referral.approvedPremisesAddress
-      isCas2Requested && isCas2Accepted -> cas2Referral.bassOffer
+      isCurfewApprovedPremisesRequired && !isCas2ApprovedPremisesRequired -> curfew?.approvedPremisesAddress
+      isCas2ApprovedPremisesRequired -> cas2Referral?.approvedPremisesAddress
+      isCas2Requested && isCas2Accepted -> cas2Referral?.bassOffer
```

Comment:
my compiler was suggesting adding this optional chaining

Reply:
As long as returning `null` here is expected rather than it blowing up, I think this is fine!

Reply:
Cool and thanks, reverted this change 😄

Reply:
Changing back again as fine either way but Gurps will have to keep changing them back 😅